### PR TITLE
OCPBUGS-25398: update Azure machine set YAML sample

### PR DESCRIPTION
**Pull request by jeana-redhat**
Version(s):
4.12+

Issue:
[OCPBUGS-25398](https://issues.redhat.com//browse/OCPBUGS-25398)

Link to docs preview:
* [Sample YAML for a compute machine set custom resource on Azure](https://72191--ocpdocs-pr.netlify.app/openshift-enterprise/latest/machine_management/creating_machinesets/creating-machineset-azure#machineset-yaml-azure_creating-machineset-azure) (standard compute)
* [Sample YAML for a compute machine set custom resource on Azure](https://72191--ocpdocs-pr.netlify.app/openshift-enterprise/latest/machine_management/creating-infrastructure-machinesets#machineset-yaml-azure_creating-infrastructure-machinesets) (infra)
* [Sample Azure provider specification](https://72191--ocpdocs-pr.netlify.app/openshift-enterprise/latest/machine_management/control_plane_machine_management/cpmso-configuration#cpmso-yaml-provider-spec-azure_cpmso-configuration) (control plane)

QE review:
- [ ] QE has approved this change.

Additional information:
* Fixed a couple other YAML issues
* Might not cherrypick clean

### CODE REVIEW

**Review by JoelSpeed — COMMENTED** [review]
/lgtm

File: modules/machineset-yaml-azure.adoc [hunk 1]
```diff
@@ -26,71 +26,71 @@ apiVersion: machine.openshift.io/v1beta1
...
           apiVersion: azureproviderconfig.openshift.io/v1beta1
           credentialsSecret:
             name: azure-cloud-credentials
             namespace: openshift-machine-api
-          image: <4>
+          image: # <4>
             offer: ""
             publisher: ""
-            resourceID: /resourceGroups/<infrastructure_id>-rg/providers/Microsoft.Compute/images/<infrastructure_id> <5>
+            resourceID: /resourceGroups/<cluster_id>-rg/providers/Microsoft.Compute/galleries/gallery_<cluster_id>/images/<cluster_id>-gen2/versions/latest # <5>
```
Comment (sunzhaohua2) [inline modules/machineset-yaml-azure.adoc:87]: seems we have been using <infrastructure_id> instead of <cluster_id>

Comment (jeana-redhat) [reply modules/machineset-yaml-azure.adoc:87]: Thanks! I'll get that one here, and fix other instances in docs in a separate PR.

**Review by kowen-rh — APPROVED** [review]
LGTM! :+1:

/remove-label peer-review-in-progress
/label peer-review-done

**Pull request merged**